src/charm.py

```diff
@@ -43,27 +43,24 @@ def __init__(self, *args):
         self._grafana_agent = COSAgentProvider(
             self,
             metrics_endpoints=[
-                {"path": "/metrics", "port": 443},
+                {"path": "/metrics", "port": HTTP_PORT},
             ],
             metrics_rules_dir="./src/prometheus_alert_rules",
             logs_rules_dir="./src/loki_alert_rules",
             dashboard_dirs=["./src/grafana_dashboards"],
             log_slots=["pollen:logs"],
         )
-        self._charm_state = None
+        self.pollen = PollenService()
+        self.hostname = self.model.get_binding("website").network.bind_address
```

Comment:
SHould this be done in the `CharmState.from_charm` method?

Reply:
This still seems to be a problem? Or do you disagree with the comment?

Reply:
Oh sorry, this is now in from_charm but i forgot to delete the line in this file. Thanks for spotting it!